pkg/sensors/tracing/genericlsm.go

```diff
@@ -0,0 +1,465 @@
+	} else if errors.Is(err, ErrMsgSyntaxLong) {
+		logger.GetLogger().WithField("policy-name", in.policyName).Warnf("TracingPolicy 'message' field too long, truncated to %d characters", TpMaxMessageLen)
+	}
+
+	tagsField, err := getPolicyTags(f.Tags)
+	if err != nil {
+		return errFn(fmt.Errorf("Error: '%v'", err))
+	}
+
+	// Parse Arguments
+	for j, a := range f.Args {
+		argType := gt.GenericTypeFromString(a.Type)
+		if argType == gt.GenericInvalidType {
+			return errFn(fmt.Errorf("Arg(%d) type '%s' unsupported", j, a.Type))
+		}
+		if a.MaxData {
+			if argType != gt.GenericCharBuffer {
+				logger.GetLogger().Warnf("maxData flag is ignored (supported for char_buf type)")
+			}
+			if !kernels.EnableLargeProgs() {
+				logger.GetLogger().Warnf("maxData flag is ignored (supported from large programs)")
+			}
+		}
+		argMValue, err := getMetaValue(&a)
+		if err != nil {
+			return errFn(err)
+		}
+		if a.Index > 4 {
+			return errFn(fmt.Errorf("Error add arg: ArgType %s Index %d out of bounds",
+				a.Type, int(a.Index)))
+		}
+		config.Arg[a.Index] = int32(argType)
+		config.ArgM[a.Index] = uint32(argMValue)
+
+		argsBTFSet[a.Index] = true
+		argP := argPrinter{index: j, ty: argType, maxData: a.MaxData, label: a.Label}
+		argSigPrinters = append(argSigPrinters, argP)
+	}
+
+	config.ArgReturn = int32(0)
+	config.ArgReturnCopy = int32(0)
+
+	// Mark remaining arguments as 'nops' the kernel side will skip
+	// copying 'nop' args.
+	for j, a := range argsBTFSet {
+		if !a {
+			if j != api.ReturnArgIndex {
+				config.Arg[j] = gt.GenericNopType
+				config.ArgM[j] = 0
+			}
+		}
+	}
+
+	config.Syscall = 0
+
+	// create a new entry on the table, and pass its id to BPF-side
+	// so that we can do the matching at event-generation time
+	lsmEntry := genericLsm{
+		config:      config,
+		argPrinters: argSigPrinters,
+		hook:        f.Hook,
+		tableId:     idtable.UninitializedEntryID,
+		policyName:  in.policyName,
+		message:     msgField,
+		tags:        tagsField,
+	}
+
+	// Parse Filters into kernel filter logic
+	lsmEntry.selectors, err = selectors.InitKernelSelectorState(f.Selectors, f.Args, nil, nil, in.selMaps)
+	if err != nil {
+		return errFn(err)
+	}
+
+	genericLsmTable.AddEntry(&lsmEntry)
+	config.FuncId = uint32(lsmEntry.tableId.ID)
+
+	lsmEntry.pinPathPrefix = sensors.PathJoin(in.sensorPath, fmt.Sprintf("glsm-%d", lsmEntry.tableId.ID))
+
+	logger.GetLogger().
+		WithField("hook", lsmEntry.hook).
+		Infof("Added lsm Hook")
+
+	return lsmEntry.tableId, nil
+}
+
+func createGenericLsmSensor(
+	spec *v1alpha1.TracingPolicySpec,
+	name string,
+	policyID policyfilter.PolicyID,
+	policyName string,
+) (*sensors.Sensor, error) {
+	var progs []*program.Program
+	var maps []*program.Map
+	var ids []idtable.EntryID
+	var selMaps *selectors.KernelSelectorMaps
+	var err error
+
+	if !kernels.MinKernelVersion("5.7") {
+		return nil, fmt.Errorf("to load LSM sensor kernel version must be greater or equal 5.7")
+	}
```

Comment:
Generally, it is a good idea to avoid doing kernel checks because in some cases features have been backported to older kernels. We have been doing it in the past, but I think we should avoid it for new code and try to migrate the code to just use feature detection.

How about we just check just for HasLSMPrograms, and we can change the error to:
> fmt.Errorf("Does you kernel support the bpf LSM? You can enable LSM BPF by modifying the GRUB configuration /etc/default/grub with GRUB_CMDLINE_LINUX=\"lsm=bpf\"")

Reply:
I agree, also I found out that rhel8 in vmtests has 4.18 kernel, but LSM BPF is enabled on this VM!